CONTRIBUTING.md

```diff
@@ -0,0 +1,513 @@
+
+### Comment Formatting
+
+Comments are to be formatted in typical `rust` style, specifically:
+
+- Use markdown to format comments.
+
+- Use the triple forward slash "///" for modules, structs, enums, traits and functions. Use double forward slash "//" for comments on individual lines of code.
+
+- Start with a high-level description of the function, adding more sentences with details if necessary.
+
+- When documenting panics, errors, or other conceptual sections, introduce a Markdown section with a single `#`, e.g.:
+
+    ```
+    # Errors
+    * ContractTooLargeError: Thrown when `contract` is larger than `MAX_CONTRACT_SIZE`.
+    ```
+
+### Content of Comments
+
+
+#### Component Comments
+
+Comments for a component (`struct`, `trait`, or `enum`) should explain what the overall
+purpose of that component is. This is usually a concept, and not a formal contract. Include anything that is not obvious about this component.
+
+**Example:**
+
+```rust
+/// The `ReadOnlyChecker` analyzes a contract to determine whether
+/// there are any violations of read-only declarations. By a "violation"
+/// we mean a function that is marked as "read only" but which tries
+/// to modify chainstate.
+pub struct ReadOnlyChecker<'a, 'b> {
+```
+
+This comment is considered positive because it explains the concept behind the class at a glance, so that the reader has some idea about what the methods will achieve, without reading each method declaration and comment. It also defines some terms that can be used in the comments on the method names.
+
+#### Function Comments
+
+The comments on a function should explain what the function does, without having to read it. Wherever practical, it should specify the contract of a function, such that a bug in the logic could be discovered by a discrepancy between contract and implementation, or such that a test could be written with only access to the function comment.
+
+Without being unnecessarily verbose, explain how the output is calculated
+from the inputs. Explain the side effects. Explain any restrictions on the inputs. Explain failure
+conditions, including when the function will panic, return an error
+or return an empty value.
+
+**Example:**
+
+```rust
+/// A contract that does not violate its read-only declarations is called
+/// *read-only correct*.
+impl<'a, 'b> ReadOnlyChecker<'a, 'b> {
+    /// Checks each top-level expression in `contract_analysis.expressions`
+    /// for read-only correctness.
+    ///
+    /// Returns successfully iff this function is read-only correct.
+    ///
+    /// # Errors
+    ///
+    /// - Returns CheckErrors::WriteAttemptedInReadOnly if there is a read-only
+    ///   violation, i.e. if some function marked read-only attempts to modify
+    ///   the chainstate.
+    pub fn run(&mut self, contract_analysis: &ContractAnalysis) -> CheckResult<()>
+```
```

Comment:
hmm something wonky is happening with the code formatting here - it ignores the ending here (` ``` `) and instead treats the following end mark as the close of the code block.

Reply:
It looks okay to me when I view the whole file: https://github.com/stacks-network/stacks-blockchain/blob/73f55526e6142d982926d9331364df11ee2b2c13/CONTRIBUTING.md

Maybe it's just the PR view that makes it go wonky?

Reply:
sure, I'd believe that.